src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -441,15 +441,20 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
 
         current_function = &x;
         std::string body;
+        bool visited_return = false;
+
         for (size_t i=0; i<x.n_body; i++) {
             this->visit_stmt(*x.m_body[i]);
+            visited_return = visited_return || (is_a<ASR::Return_t>(*x.m_body[i]));
             body += src;
         }
         current_function = nullptr;
```

Comment:
What if there are some statements after the return, and reachable via goto? Not recommended, but possible.

How about this:
```
bool visited_return = false;
if (x.n_body > 0 && is_a<ASR::Return_t>(*x.m_body[x.n_body-1])) visited_return = true;
```
That way if the last statement was a return, then we don't repeat it, otherwise we add a return.

Reply:
That sounds good.